### PR TITLE
Avoid blank background urls and src attributes

### PR DESCRIPTION
**Pull request by andykent**
See https://github.com/imgix/react-imgix/issues/49 for the full context.

I've added some additional tests and they are all passing but I've not actually tried running the built code in a real project yet so I can't be sure there aren't any regressions.

### CODE REVIEW

**Comment by paulstraw** [comment]
Looked through this, and it makes sense to me. I'll pull down shortly to verify in a test project.

**Comment by paulstraw** [comment]
Finally got a chance to try this out, and it seems to be working as expected. @andykent, have you been using this branch? Run into anything? If @frederickfogerty is good with it, I'm happy to merge this.

File: src/index.test.js [hunk 1]
```diff
@@ -31,6 +31,20 @@ describe('<img> mode', () => {
...
 })
+describe('<img> mode - pre-mount', () => {
+  beforeEach(() => {
+    tree = sd.shallowRender(
+      <Imgix
+        src={src}
+      />
+    )
+    vdom = tree.getRenderOutput()
+  })
```
Comment (frederickfogerty) [inline src/index.test.js]: Is this emulating the pre-mount state because `tree.getMountedInstance()` is not present?

File: src/index.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ import processImage from './support.js'
 
 const roundToNearest = (size, precision) => precision * Math.ceil(size / precision)
 
+const isPresent = (str) => str && typeof str === 'string' && str.length > 0
```
Comment (frederickfogerty) [inline src/index.js]: Could we rename to `isStringNotEmpty`?

File: src/index.js [hunk 3]
```diff
@@ -85,8 +87,8 @@ export default class ReactImgix extends Component {
       src,
       ...other
     } = this.props
-    let _src = ''
-    let srcSet = ''
+    let _src = null
+    let srcSet = null
```
Comment (frederickfogerty) [inline src/index.js]: 👍

**Comment by andykent** [comment]
Apologies for the delay. We just had a baby girl so I was off on paternity leave but back now. I've made the requested change and also inlined the conditional as I think it makes the  behaviour more obvious to parse now. 

I'll test if out on our site today and report back shortly but I think we are good.

**Comment by andykent** [comment]
Yep, this is working for us in our codebase.

@frederickfogerty are you able to re-review and merge this in when you get a moment.

It would be really great for us to be able to switch back to an official release sometime soon.

**Comment by frederickfogerty** [comment]
Hi @andykent I've just added a comment to the test answering my own question. I'm going to merge this now, but can you please review that comment, and if it's wrong, I can fix that up on master.

**Pull request merged**